llvm/lib/Analysis/CaptureTracking.cpp

```diff
@@ -438,18 +448,21 @@ void llvm::PointerMayBeCaptured(const Value *V, CaptureTracker *Tracker,
   };
   while (!Worklist.empty()) {
     const Use *U = Worklist.pop_back_val();
-    switch (DetermineUseCaptureKind(*U, IsDereferenceableOrNull)) {
-    case UseCaptureKind::NO_CAPTURE:
+    CaptureInfo CI = DetermineUseCaptureKind(*U, IsDereferenceableOrNull);
+    if (capturesNothing(CI))
       continue;
-    case UseCaptureKind::MAY_CAPTURE:
-      if (Tracker->captured(U))
+    CaptureComponents RetCC = CI.getRetComponents();
+    if (!capturesNothing(CI.getOtherComponents())) {
```

Comment:
Why is this not `capturesNothing(CI.getOtherComponents())`? As I understand this for the previous `PASSTHROUGH` case which should be `other:none, ret:<not none>`.

Reply:
This is the MAY_CAPTURE case. The PASSTHROUGH case is the one below (AddUses).

Reply:
This case falls through to the the PASSTHROUGH logic though. Or is it made equivilent because `captured(...)` only returns `nullopt` or `none`?

Reply:
It falls through in the case where we handle something like `captures(address_is_null, ret: address, provenance)`. Depending on the captured() return value, we'll fall through to handling the `ret` part as passthrough.

Reply:
I think this code should be clearer now that it uses the enum return value.